File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& arr) {
    int target= arr.size()/2;
    int count=0;
    int element;
    for(int i=0;i<arr.size();i++){
        if(count==0) {
            count=1;
            element =arr[i];
        }
        else if(arr[i]==element) count++;
        else if(arr[i]!=element) count--;
    
   
}

//now check if element that is in majority is > target or not
    int count1=0;
    for(int i=0;i<arr.size();i++){
        if(arr[i]==element) count1++;
       
    }
     if(count1>target) return element;
     else return -1;
}
    
};

File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        count=0
        n=len(nums)
        target=n/2
        element=-1
        for i in range(n):
            if count==0:
                count=1
                element=nums[i]
            elif nums[i]==element:
                count+=1
            else:
                count-=1
        

        count1=0
        for i in range(n):
            if nums[i]==element:
                count1+=1
        if count1>=target:
            return element
        return -1
        
        